React Router 설정, Sidebar, dummyChart

EarlyHail opened the pull request:
### 구현의도
- React Router 설정
- Sidebar 추가
- Dummy Chart 추가

### 기능 흐름도, 클래스 다이어그램(선택사항)
- 

### 사용된 기술(선택사항)
- 

### 리뷰 & 논의사항 & 궁금한점(선택사항)
-

EarlyHail committed:
[add] React Router 설정

- Browser Router를 이용한 Route 설정
  기본 페이지 (Issue, IssueList, Discover) 테스트 component 생성
  라우팅 테스트 완료

EarlyHail committed:
[refactor] React Router 변경

- IssueList --> Issue로 변경
  기존 IssueList 삭제
- Issue --> IssueDetail로 변경
   IssueDetail 추가

EarlyHail committed:
[add] Sidebar 추가

- Materia UI 기반 Sidebar 추가
  `@Material-ui/core/icons` 추가
- default CSS 추가
  `@Material-ui/core/CssBaseline` 추가 (index.tsx)

EarlyHail committed:
[add] issue에 dummy chart 추가

- Billboard를 이용한 dummy chart 추가

EarlyHail committed:
[refactor] dummy Chart 리팩토링

- dummy chart 코드를 `issue.tsx`에서 `components/Chart`로 분리

EarlyHail committed:
[add] issue 상세 페이지 연결

- `issue.tsx`에 이슈 상세 페이지 링크 추가

junsushin-dev commented on src/Router.tsx:
```diff
@@ -0,0 +1,32 @@
+//   path="/new-issue"
+//   exact={true}
+//   component={NewIssuePage}
+// />
```
Comment 처리된 과거 코드가 보이는데 삭제하면 더 좋을 것 같습니다! 😀

junsushin-dev commented on src/components/layout/Sidebar/index.tsx:
```diff
@@ -0,0 +1,125 @@
+        </Box>
+      </Box>
+      <Box display="flex" flexDirection="column" pt={2}>
+        <Tab to="/project" activeClassName={style.activeStyle}>
+          <Box display="flex" alignItems="center" px={3}>
+            <ArchiveRoundedIcon />
+            <Box px={2}>Projects</Box>
+          </Box>
+        </Tab>
+        <Tab to="/issue" activeClassName={style.activeStyle}>
+          <Box display="flex" alignItems="center" px={3}>
+            <AssignmentIcon />
+            <Box px={2}>Issues</Box>
+          </Box>
+        </Tab>
+        <Tab to="/discover" activeClassName={style.activeStyle}>
+          <Box display="flex" alignItems="center" px={3}>
+            <HighlightIcon />
+            <Box px={2}>Discover</Box>
+          </Box>
+        </Tab>
+        <Tab to="/alerts" activeClassName={style.activeStyle}>
+          <Box display="flex" alignItems="center" px={3}>
+            <ReportProblemIcon />
+            <Box px={2}>Alerts</Box>
+          </Box>
+        </Tab>
```
이 부분은 나중에 Page Class를 따로 정의해서 함수형으로 리팩토링 해볼 수도 있을 것 같네요! 
Class Instance에 path, component, text 를 넣어놓고 pages.map((page) => ...) 이런 식이면 될 것 같아요. 
지금은 프로토타이핑이라 놔두어도 괜찮고 나중에 두번째 이터레이션 쯤에 한번에 해보면 좋을 것 같습니다. ☺

juyoungpark718 replied:
중복되는 부분이라 준수님 말씀하신 것처럼 따로 컴포넌트로 분리하는게 훨씬 좋아보입니다.

junsushin-dev commented on src/pages/Issue.tsx:
```diff
@@ -1,7 +1,133 @@
+      else data.push({ occuredAt: issue.occuredAt, count: 1 });
+    }
+    return data;
+  }, []);
```
이쪽 Logic은 암묵적으로 issues 데이터가 시간 순으로 정렬되었다는 것을 가정하고 있는 것 같습니다. 혹시 추후에 비슷하게 재사용하실 예정이라면, 주석을 추가하거나 변수명을 issuesOrderedByTime 등으로 바꾸어 주면 더 좋을 것 같아요! 😉

EarlyHail replied:
이 로직이 제 기억에는 서버에서 시간순으로 데이터가 오지 않을수도 있어서 정렬을 해주는거라고 들었습니다. 

이 부분은 나중에 차트 데이터 api를 만들면서 보완을 할지 쓰면 될지 결정하면 될것같고
일단은 이름을 좀 더 명확하게 sortChartData()로 변경했습니다.

@juyoungpark718 혹시 다른 의견있으시면 남겨주세요!

juyoungpark718 replied:
시간순으로 정렬된다고 가정하고 진행했던 데이터가 맞습니다. 추후에 준수님 의견으로 리팩토링을 진행하면 더 좋을 것 같아요.
@EarlyHail  해당 부분은 issue가 생성된 시간으로 해당 issue를 묶는 함수였습니다. (해당 시간에 발생한 issue를 그래프에 표현하기 위한)

junsushin-dev reviewed (commented):
Frontend 세팅하느라 정말 고생 많으셨습니다! 특히 라우터 잡는 쪽이 고생이 많으셨을 것 같습니다. 논외로 commit 메시지의 tag가 [add] 로 되어있는 부분이 있는데, 저희 Git 활용 그라운드 룰에 맞추어서 다음부터 [feat]으로 해주시면 더 좋을 것 같습니다. 😊

EarlyHail committed:
[refactor] 주석 제거, 함수명 변경

- 개발 과정에서 남겨놨던 주석제거
   `Router.tsx`
- 함수이름 변경
   `components/Chart/index.tsx`
   Data 정렬함수 `getChartData` -> `sortChartData`로 변경

juyoungpark718 reviewed (approved):
고생하셨습니다! 👍

EarlyHail committed:
[refactor] 함수명 변경

- `sortChartData`를 기능에 맞게 `issuesOrderedByTime`으로 변경

junsushin-dev reviewed (approved):
코멘트 사항 정말 광속으로 반영해주셨군요... 감사합니다. 호진님 주영님 고생하셨습니다!! 😁

saeeng commented on src/App.tsx:
```diff
@@ -1,22 +1,10 @@
-          Learn React
-        </a>
-      </header>
+      <MainLayout />
```
라우터 설정을 레이아웃 내부에서 넣어주는것보다 엔트리 부분에서 넣어주는건 어떨까요?
레이아웃 파일은 구조를 잡는다는 역할이 조금은 더 명확해 질것같습니다.

saeeng reviewed (approved):
좋습니다!!
